src/main/java/org/datadog/jenkins/plugins/datadog/apm/GolangConfigurator.java

```diff
@@ -0,0 +1,51 @@
+package org.datadog.jenkins.plugins.datadog.apm;
+
+import hudson.FilePath;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.datadog.jenkins.plugins.datadog.steps.TestOptimization;
+
+public class GolangConfigurator implements TracerConfigurator {
+
+    private static final int SHELL_COMMAND_TIMEOUT_MILLIS = 180_000;
+
+    @Override
+    public Map<String, String> configure(TestOptimization testOptimization, Node node, FilePath workspacePath, Map<String, String> envs, TaskListener listener) throws Exception {
+        if (!workspacePath.child("go.mod").exists()) {
+            // the workspace is not a golang project?
+            return Collections.emptyMap();
+        }
```

Comment:
`go.mod` was introduced since `Go v1.11` in 2018. Probably using that version is something very uncommon now, but I guess we should include a minimum Go version required in order to use the autoinstrumentation correctly.

Reply:
Good point, added a version check